Cargo.toml

```diff
@@ -1,11 +1,11 @@
 [workspace]
-members = ["felt", "merkle", "poseidon", "randomness", "channel"]
+members = ["felt", "merkle", "poseidon", "randomness", "channel", "fri"]
 resolver = "2"
 
 [workspace.dependencies]
 anyhow = "1.0.86"
-ark-ff = "0.4.2"
-ark-poly = "0.4.2"
+ark-ff = { git = "https://github.com/jaehunkim/algebra", branch = "v0.4.2-fix" }
+ark-poly = { git = "https://github.com/jaehunkim/algebra", branch = "v0.4.2-fix" }
```

Comment:
diff : https://github.com/arkworks-rs/algebra/compare/master...jaehunkim:algebra:master
When q is 2, the function get_root_of_unity(n) always returns None for any n greater than 1.
I have temporarily patched this issue for now.

```
let q_adicity = crate::utils::k_adicity(q, n);
let q_part = q.checked_pow(q_adicity)?;
let two_adicity = crate::utils::k_adicity(2, n);
let two_part = 2u64.checked_pow(two_adicity)?;
if n != two_part * q_part
    || (two_adicity > Self::TWO_ADICITY)
    || (q_adicity > small_subgroup_base_adicity)
{
    return None;
}
```